src/contract/contract.ts

```diff
@@ -0,0 +1,62 @@
+//eslint-disable-next-line
+require("dotenv").config();
+import Web3 from "web3";
+import { keccak256 } from "js-sha3";
+import { AbiInput } from "web3-utils";
+import { Log } from "web3-core";
+
+export const keccakTopic = (topic: string) => "0x" + keccak256(topic);
+
+export interface ContractResult {
+  contractAddress: string;
+  contractName: string;
+  blockNumber: number;
+  blockHash: string;
+}
+
+export const DSNPMigrationABI: AbiInput[] = [
+  {
+    indexed: false,
+    internalType: "address",
+    name: "contractAddr",
+    type: "address",
+  },
+  {
+    indexed: false,
+    internalType: "string",
+    name: "contractName",
+    type: "string",
+  },
+];
+
+const decodeReturnValues = (web3: Web3, inputs: AbiInput[], topic: string, logs: Log[]) => {
```

Comment:
perhaps we can use "named parameters"

Reply:
we had gone back and forth about this a while ago. and I think the decision was to be explicit for now?